doc/design/if_else_op.md

```diff
@@ -0,0 +1,59 @@
+IfOp should have only one branch. An IfOp operator takes a `cond` variable whose value must be a vector of N boolean elements. Its return value has M (M<=N) instances, each corresponds to a true element in `cond`.
```

Comment:
should we add a static IfElseOp just like TF, just run one branch.

This design describe a DynamicIfElseOp ? am I right?

Reply:
Let's make this IfElseOp. If you are going to add a static conditional branching structure later, we can name it StaticIfElseOp.